src/pages/spawn/spawnHost/SpawnHostModal.tsx

```diff
@@ -154,6 +156,15 @@ export const SpawnHostModal: React.VFC<SpawnHostModalProps> = ({
     });
   }, [distroId, dispatch, distrosData?.distros, unexpirableCountReached]);
 
+  const removeModalQueryParam = useCallback(() => {
+    setSearchParams(
+      [...searchParams.entries()].filter(([k]) => k !== "spawnHost"),
```

Comment:
We should probably extract `spawnHost` into an enum and reuse it here and [here](https://github.com/evergreen-ci/spruce/blob/5f95e9396419a573b210db72d561b6d89fc41a46/src/pages/spawn/spawnHost/SpawnHostButton.tsx#L24-L25)